source/copy-as-markdown.js

```diff
@@ -74,11 +84,15 @@ browser.contextMenus.onClicked.addListener(async (info, tab) => {
 
 	const markdownData = turndownService.turndown(htmlContent);
 
-	const inputElement = document.createElement('textarea');
-	document.body.append(inputElement);
-	inputElement.value = markdownData;
-	inputElement.focus();
-	inputElement.select();
-	document.execCommand('Copy');
-	inputElement.remove();
+	copy(markdownData);
+});
+
+// Listener for events from shortcuts
+browser.commands.onCommand.addListener(async command => {
+	if (command === 'copy-selection-as-md') {
+		const completionData = await browser.tabs.executeScript({code: __INJECTIBLE_CODE__});
+		const htmlContent = completionData[0] || '';
+		const markdownData = turndownService.turndown(htmlContent);
+		copy(markdownData);
```

Comment:
This is the same thing as I see in the other event listener, maybe move that into a different function and see if that works.

I vaguely remember that `browser.tabs.executeScript` has to be called in the same event listener callback or it won't work in Firefox, in which case this would be working as expected. Just want you to give a test and confirm, thanks!

Reply:
I tested it on firefox, and it seems to be fine. It's mostly the same as the code in `browser.contextMenus.onClicked.addListener`, but maybe it's not enough for us to move it into another function, I think, because it's not that complex, another reason is, there is some code I don't know how it works, and I haven't concentrated on extension development for a while, so, I would appreciate if you can help to improve when necessary. Thanks.